docs/lib/storage/fragments/android/configureaccess.md

```diff
@@ -1,28 +1,196 @@
+ When adding the Storage category, you configure the level of access authenticated and guest users have to your S3 bucket. Additionally, when uploading files using the Storage category, you can specify the access level for that file to be either public, protected, or private.
+
+- **Public** Accessible by all users
+- **Protected** Readable by all users, but only writable by the creating user
+- **Private** Readable and writable only by the creating user
+
+For protected and private access, the `user_id` in the prefix corresponds to the unique ID for the creating user.
+
+<amplify-callout>
+
+The default access level for the Storage category is **public**. Unless you specify otherwise, all uploaded files will be publicly available for all users.
+
+</amplify-callout>
+
 ## Protected access
 
-Create an options object with the protected access level to restrict access for certain objects.
+Create an options object specifying the protected access level to allow other users to read the object:
+
+<amplify-block-switcher>
+<amplify-block name="Java">
 
 ```java
-  StorageUploadFileOptions options =
-    StorageUploadFileOptions.builder()
-    .accessLevel(StorageAccessLevel.PROTECTED)
-    .build();
+private void uploadFile(String key, File file) {
+    StorageUploadFileOptions options =
+            StorageUploadFileOptions.builder()
+                    .accessLevel(StorageAccessLevel.PROTECTED)
+                    .build();
+
+    Amplify.Storage.uploadFile(
+            key,
+            file,
+            options,
+            result -> Log.i("MyAmplifyApplication", "Successfully uploaded: " + key),
+            error -> Log.e("MyAmplifyApplication", "Upload failed", error)
+    );
+}
+```
 
-  Amplify.Storage.uploadFile(uploadFileName, localFilePath, options, resultListener) { ... }
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+private fun uploadFile(key: String, file: File) {
+    val options = StorageUploadFileOptions.builder()
+        .accessLevel(StorageAccessLevel.PROTECTED)
+        .build()
+    
+    Amplify.Storage.uploadFile(
+        key,
+        file,
+        options,
+        { Log.i("MyAmplifyApplication", "Successfully uploaded: $key" )},
+        { error -> Log.e("MyAmplifyApplication", "Upload failed", error)}
+    )
+}
 ```
 
-Another user that wants to read the file can specify the user that created it:
+</amplify-block>
+</amplify-block-switcher>
+
+For other users to read the file, you must specify the user ID of the creating user in the passed options:
```

Comment:
Should we include a note about where to find this ID?

Reply:
Hmm, open to suggestions! I'd think this would vary depending on the developers' application details.

Reply:
Hm yea, let's punt it?